compiler/backend/src/org/jetbrains/kotlin/codegen/optimization/boxing/BoxingInterpreter.kt

```diff
@@ -147,6 +151,16 @@ open class BoxingInterpreter(
                 value is CleanBoxedValue ||
                 value.type != null && isProgressionClass(value.type)
 
+    private fun isCastToProgression(insn: AbstractInsnNode): Boolean {
+        // insn can only be of type TypeInsnNode, so the type conversion here is safe
```

Comment:
Minor: better add an assertion that `insn.opcode == Opcodes.CHECKCAST` or something like that.